### ISSUE TITLE
💅 `noImportAssign` misses writes to namespace import members

### ISSUE THREAD

**Issue by AlbinoGeek** [biomejs/biome#11948]
### Environment information

<details>

```bash
CLI:
  Version:                      2.5.14

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_DISTRIBUTION:           npm
  JS_RUNTIME_NAME:              node
```

</details>

### Rule name

`suspicious/noImportAssign`

### Playground link

https://biomejs.dev/playground/?lintRules=noImportAssign#files.noImportAssign.js=aQBtAHAAbwByAHQAIAAqACAAYQBzACAAbgBzACAAZgByAG8AbQAgACIALgAvAG0AbwBkAC4AagBzACIAOwAgAC8ALwAgAG0AbwBkAC4AagBzADoAIABlAHgAcABvAHIAdAAgAGwAZQB0ACAAYQAgAD0AIAAxADsACgBlAHgAcABvAHIAdAAgAGYAdQBuAGMAdABpAG8AbgAgAGYAKAApACAAewAKACAAIABuAHMAIAA9ACAAewB9ADsAIAAvAC8AIAByAGUAcABvAHIAdABlAGQACgAgACAAbgBzAC4AYQAgAD0AIAAxADsAIAAvAC8AIABuAG8AdAAgAHIAZQBwAG8AcgB0AGUAZAAsACAAVAB5AHAAZQBFAHIAcgBvAHIAIABhAHQAIAByAHUAbgB0AGkAbQBlAAoAIAAgAGQAZQBsAGUAdABlACAAbgBzAC4AYQA7ACAALwAvACAAbgBvAHQAIAByAGUAcABvAHIAdABlAGQALAAgAFQAeQBwAGUARQByAHIAbwByACAAYQB0ACAAcgB1AG4AdABpAG0AZQAKACAAIABPAGIAagBlAGMAdAAuAGEAcwBzAGkAZwBuACgAbgBzACwAIAB7ACAAYQA6ACAAMgAgAH0AKQA7ACAALwAvACAAbgBvAHQAIAByAGUAcABvAHIAdABlAGQALAAgAFQAeQBwAGUARQByAHIAbwByACAAYQB0ACAAcgB1AG4AdABpAG0AZQAKAH0ACgA%3D

```js
import * as ns from "./mod.js"; // mod.js: export let a = 1;
export function f() {
  ns = {}; // reported
  ns.a = 1; // not reported, TypeError at runtime
  delete ns.a; // not reported, TypeError at runtime
  Object.assign(ns, { a: 2 }); // not reported, TypeError at runtime
}
```

Module namespace objects are frozen, so writing to, deleting, or `Object.assign`ing an exported member throws `TypeError` in every ES module (checked in Node with `mod.js` exporting `a`). Only reassigning the `ns` binding is reported. ESLint's `no-import-assign` reports all four lines ("The members of 'ns' are read-only").

Reproduced with the 2.5.14 CLI and in the playground (f74eb38).

### Expected result

Assignments, compound assignments, updates, `delete`, and mutating built-ins (`Object.assign`, `Object.defineProperty`, `Reflect.set`, ...) targeting a member of a namespace import are reported.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

---

Tool Use Disclaimer: Claude Code using Anthropic Opus 5.5 helped build the reproduction.
